tests/test_interval.py

```diff
@@ -47,10 +48,10 @@ def test_interval_length() -> None:
 
     assert_type(idres, "pd.Interval[pd.Timestamp]")
     if TYPE_CHECKING:
-        20 in i1  # type: ignore
-        i1 + pd.Timestamp("2000-03-03")  # type: ignore
-        i1 * 3  # type: ignore
-        i1 * pd.Timedelta(seconds=20)  # type: ignore
+        20 in i1
```

Comment:
Removing this `type: ignore` means we think this code is legal, but it shouldn't be.  You can't check that an `int` is contained inside of an `Interval[Timestamp]`, so the type checker should pick that up.

I'm hesitant to make this change unless we can fix the stub to error out on this line.

I'm pretty sure at one point it did used to pick that up - not sure what changed.

Reply:
I think it will take quite some effort to remove annotations like these
```py
def __add__(self, other: TimestampSeries) -> TimestampSeries: ...
def __add__(self, other: Timestamp) -> TimestampSeries: ..
```
but also allow legit combinations.